tests/unit/utils/limiters/test_adjustable_semaphor.py

```diff
@@ -119,10 +128,11 @@ async def semaphore_worker(worker_id: int):
             logger.info(f"Current iteration {idx}/{len(state.limits)}")
             await asyncio.sleep(get_sleep_time(sleep_interval))
             assert semaphore.waiting == waiting_count
+            assert semaphore.processing >= 0
+            assert semaphore.processing == state.tasks - waiting_count - len(final_execution_order)
             logger.info(f"Changing max limit from {semaphore.limit} to {new_limit}")
             semaphore.change_max_limit(new_limit)
             assert semaphore.limit == new_limit
```

Comment:
we could add optional checking for _value:
```suggestion
            assert semaphore.limit == new_limit
            if state.expected_internal_values:
                assert semaphore._value == state.expected_internal_values[idx]
```

and test-case:
```python
            TestState(
                tasks=3,
                limits=[0, 5, 4, 3, 2, 3, 4, 5],
                ticks=[1, 1, 1, 1, 1, 1, 1, 1],
                execution_time=[10, 10, 10],
                waiting_counts=[3, 3, 0, 0, 0, 0, 0, 0],
                final_order=[0, 1, 2],
                expected_internal_values=[0, 2, 1, 0, 0, 0, 1, 2],
            ),
```

Reply:
I think we should not test the implementation.